### DISCUSSION TITLE
Add documentation for conventions/best practices for handling packages

### DISCUSSION THREAD

**Discussion by russellbanks** [microsoft/winget-pkgs#78491] [General]
There are often cases when different contributors have varying outlooks on how best packages' manifests and versions should be set up. Therefore, it makes sense to have documentation on conventions/best practices for handling packages.

For example:
- If a source has an `exe` and an `msi`, do you provide both or just the `msi`? (Answer: provide both - https://github.com/microsoft/winget-pkgs/discussions/68595#discussioncomment-3325537)
- If a source has an installer and a portable exe, do you provide both or just the installer? (Answer: provide both - https://github.com/microsoft/winget-pkgs/discussions/68595#discussioncomment-3351587)
- Should rich version histories be provided or latest version only? (Answer: https://github.com/microsoft/winget-pkgs/discussions/66854#discussioncomment-3188217)
   - Does this vary depending on the package (E.g. Nightlies might only want to provide the latest version, but stables should contain every version)? (Answer: "Even for nightlies, there are many reasons to keep all versions (as long as the URLs are still valid) such as Regression testing, dependency validation, etc.")
- What are acceptable cases for a package to be deleted? (Answer: Installer is no longer available or the publisher does not want their package on winget - may be other cases not covered)
   - What are non-acceptable cases for packages to be deleted? (E.g. CVEs)
- Should packages be split when a package is receiving two different channels of updates (I.e. 7.x.x and 6.x.x get updates)? (Answer: Not certain, but at this stage, I imagine it's best to wait for package pinning)
- Should arbitrarily old versions be removed if a package's manifests haven't been maintained in a long time? (Answer: probably not)

This list is not exhaustive and the "answers" I have described are mostly what I have learned myself from the opinions of moderators and from questions asked in discussions.

This should provide more clarity for contributors and provide more of a justification for PRs to be accepted/rejected based on whether they follow/don't follow the repositories' conventions.

**Comment by vedantmgoyal9** [comment]
@denelon https://github.com/microsoft/winget-pkgs/labels/Issue-Docs

**Comment by denelon** [comment]
I've converted this to a discussion so we can get multiple opinions. I would lean in the direction of what is best for the community long term.

**Comment by Trenly** [comment]
I think the answers here in terms of "What's best for the long term" are pretty straightforward. I'll post a condensed version of my thoughts here

* What should we do if there are multiple installer types for the same application? e.g - MSI / EXE / Portable
  * Provide as many installer types as possible for completeness
*  When should packages be removed?
   * Only when the manifest is "broken". If the Installer URLs are 404, if there is PUA packaged with the installer, if the installer doesn’t meet repository policies. Alternatively, publishers requesting their package be removed may qualify for a reason for removal, if the publisher themselves makes the request sua sponte.
* When shouldn’t packages be removed?
  * Packages should not be removed for Arbitrary reasons, because there are "a lot of versions", CVE's, being outdated, or being un-maintained. Winget is still developing, and there is value in keeping any version of software which still works. There are requests on the backlog to help inform users of CVE's, unmaintained software, etc.
* Should packages be split when receiving separate channels of updates?
  * It depends, but generally, yes. If two versions are maintained separately, they should very likely be split - especially if they install side-by-side. The idea is that a user may not want to switch from a v1 app to a v2 app, but would still want all the latest updates to v1. 
* What is the ideal PackageIdentifier when splitting packages (and in general)
  * The best way to split packages is `Publisher.Package.MajorChannel.MinorChannel.OtherChannel.Architecture.InstallerType` where fields that are unused are ommitted. For example, `Corsair.iCUE.4 - Publisher.Package.MajorChannel` or `Python.Python.3.9 - Publisher.Package.MajorChannel.MinorChannel` or `EclipseAdoptium.Temurin.20.JDK - Publisher.Package.MajorChannel.OtherChannel`. Note that not all packages currently follow this convention, but I suggest it as a best practice for package IDs
* When should packages be moved?
  * Any time a package is completely moved to a new identifier, a discussion should be created here before the package is actually moved. Packages may need to be moved when splitting between channels, when there are two different identifiers being used for the same publisher, or when the Identifier needs to be changed for clarity. If a package is being published under a new name, it should be redirected instead using the two-manifest method described in the FAQ
